src/dbnode/storage/shard_insert_queue.go

```diff
@@ -203,9 +203,7 @@ func (q *dbShardInsertQueue) insertLoop() {
 		}
 		// Batch together for single insertion.
 		for _, batchByCPUCore := range batch.insertsByCPUCore {
-			batchByCPUCore.Lock()
```

Comment:
this also looks like unnecessary locking because we update `batch` only while doing `Rotate` few lines before and `batch` itself is local to `insertLoop`.